frontend/src/components/pages/teacher/ClassroomsPage.tsx

```diff
@@ -1,8 +1,210 @@
+  GridItem,
+  HStack,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+
+import { Grade } from "../../../APIClients/types/UserClientTypes";
+import DisplayAssessmentsIllustration from "../../../assets/illustrations/display-assessments.svg";
+import { TabEnum } from "../../../types/AuthTypes";
+import ClassroomCard from "../../classrooms/ClassroomCard";
+import MessageContainerClassroom from "../../common/MessageContainerClassroom";
+import Pagination from "../../common/table/Pagination";
+import AddClassroomModal from "../../user-management/student/AddClassroomModal";
 
 const ClassroomsPage = (): React.ReactElement => {
-  return <Text>Placeholder for Classrooms page.</Text>;
+  const unselectedTabColor = "#727278";
+  const [tabIndex, setTabIndex] = React.useState<TabEnum>(TabEnum.ACTIVE);
+  const methods = useForm();
+
+  const classrooms = [
+    {
+      id: "1",
+      name: "David",
+      studentCount: 100,
+      assessmentCount: 100,
+      grade: Grade.GRADE_4,
+      activeAssessments: 100,
+    },
+    {
+      id: "2",
+      name: "David 2",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "3",
+      name: "David 3",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "4",
+      name: "David 4",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "5",
+      name: "David 5",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "6",
+      name: "David 6",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "7",
+      name: "David 7",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "8",
+      name: "David 8",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "9",
+      name: "David 9",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "10",
+      name: "David 10",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+  ];
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 8;
+  const totalItems = classrooms.length;
+  const totalPages = Math.ceil(totalItems / itemsPerPage);
+
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = Math.min(startIndex + itemsPerPage, totalItems);
+  const itemsToShow = classrooms.slice(startIndex, endIndex);
```

Comment:
Instead of doing the calculation manually, consider using the usePaginatedData hook!